docs/snapcraft-parts.md

```diff
@@ -75,6 +75,36 @@ this in the snap directory, which is the final layout for the snap, this is
 where everything should look clean and crisp for a good quality snap.
 
 
+## Mastering the file system layout
+
+Snaps are typically deployed under `/snaps`, but because this location might
+change or multiple installation locations might be used (e.g.
+`/pre-installed` and `/snaps`), it's best not to hardcode it: to relate to
+resources shipped in your snap, use the `SNAP_DATA` environment variable.
+
+The contents of installed snaps is read-only; to store application state and
+data, a system-wide and per-user directories are made-available to snaps: use
+the `SNAP_DATA` environment variable to relate to the system-wide state
+directory for your snap and the `SNAP_USER_DATA` environment variable for the
+per-user directory.
+
+The layout of files under the `SNAP` is up to application authors, but it's
+quite common to use `lib/` and `bin/` for application runtime libraries and
+binaries, respectively. If your snap isn’t architecture-independent (such as
+a script), it's best to use directories qualified with the architecture such
+as `lib/arm-linux-gnueabihf/` or `bin/x86_64-linux-gnu/`.
```

Comment:
I'm curious about the reasoning behind this. For .debs it makes sense since everything is installed alongside each other, but I feel like it makes less sense in a .snap.